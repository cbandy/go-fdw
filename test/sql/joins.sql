-- vim: set expandtab shiftwidth=2 syntax=pgsql tabstop=2 :

CREATE SERVER none FOREIGN DATA WRAPPER go_test_fdw;

CREATE FOREIGN TABLE one (id integer, col1 text)
SERVER none OPTIONS (test 'fdw_join');

CREATE FOREIGN TABLE two (id integer, col2 text)
SERVER none OPTIONS (test 'fdw_join');

--
DO $$ BEGIN RAISE INFO 'local join'; END $$;
--
SELECT * FROM one JOIN (VALUES (1, 'a'), (2, 'b'), (3, 'c')) x (id, value) USING (id) ORDER BY id;

--
DO $$ BEGIN RAISE INFO 'fdw self-join'; END $$;
--
SELECT * FROM one a JOIN one b USING (id) ORDER BY id;

--
DO $$ BEGIN RAISE INFO 'fdw inner join'; END $$;
--
SELECT * FROM one JOIN two USING (id) ORDER BY id;

--
DO $$ BEGIN RAISE INFO 'fdw left join'; END $$;
--
SELECT * FROM one LEFT JOIN two USING (id) ORDER BY id;

--
DO $$ BEGIN RAISE INFO 'fdw right join'; END $$;
--
SELECT * FROM one RIGHT JOIN two USING (id) ORDER BY id;
