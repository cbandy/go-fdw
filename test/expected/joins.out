-- vim: set expandtab shiftwidth=2 syntax=pgsql tabstop=2 :
CREATE SERVER none FOREIGN DATA WRAPPER go_test_fdw;
CREATE FOREIGN TABLE one (id integer, col1 text)
SERVER none OPTIONS (test 'fdw_join');
CREATE FOREIGN TABLE two (id integer, col2 text)
SERVER none OPTIONS (test 'fdw_join');
--
DO $$ BEGIN RAISE INFO 'local join'; END $$;
INFO:  local join
--
SELECT * FROM one JOIN (VALUES (1, 'a'), (2, 'b'), (3, 'c')) x (id, value) USING (id) ORDER BY id;
 id | col1 | value 
----+------+-------
  1 | x    | a
  2 | y    | b
  3 | z    | c
(3 rows)

--
DO $$ BEGIN RAISE INFO 'fdw self-join'; END $$;
INFO:  fdw self-join
--
SELECT * FROM one a JOIN one b USING (id) ORDER BY id;
 id | col1 | col1 
----+------+------
  1 | x    | x
  2 | y    | y
  3 | z    | z
(3 rows)

--
DO $$ BEGIN RAISE INFO 'fdw inner join'; END $$;
INFO:  fdw inner join
--
SELECT * FROM one JOIN two USING (id) ORDER BY id;
 id | col1 | col2 
----+------+------
  2 | y    | k
  3 | z    | l
(2 rows)

--
DO $$ BEGIN RAISE INFO 'fdw left join'; END $$;
INFO:  fdw left join
--
SELECT * FROM one LEFT JOIN two USING (id) ORDER BY id;
 id | col1 | col2 
----+------+------
  1 | x    | 
  2 | y    | k
  3 | z    | l
(3 rows)

--
DO $$ BEGIN RAISE INFO 'fdw right join'; END $$;
INFO:  fdw right join
--
SELECT * FROM one RIGHT JOIN two USING (id) ORDER BY id;
 id | col1 | col2 
----+------+------
  2 | y    | k
  3 | z    | l
  4 |      | m
  5 |      | n
(4 rows)

